rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/LogoutResource.java

```diff
@@ -75,19 +67,80 @@ public Response logoutUser(String user) {
+                entity("User requested to log out does not match actual user logged in").build();
+        }
+    }
+
+    @Override
+    public Response logout(String unauthorize, String requestedUser) {
+        MultiSessionAttributeAdapter session = MultiSessionAttributeAdapter.of(req, false);
+        WebEntitlementContext ctx = (WebEntitlementContext) Entitlements.getEntitlementContext();
+        String currentUser = ctx==null ? null : ctx.user();
+        log.debug("Logging out: {}, session id {} ({})"+", unauthorized={}", currentUser, (session!=null ? session.getId()+" " : ""), session, unauthorize);
+        
+        MutableMap<String,String> body = MutableMap.of();
+        body.addIfNotNull("currentUser", currentUser);
+        body.addIfNotNull("requestedUser", requestedUser);
+        body.addIfNotNull("sessionId", session==null ? null : session.getId());
+        body.addIfNotNull("requestedSessionId", req.getRequestedSessionId());
+        
+        if (requestedUser!=null && !requestedUser.equals(currentUser)) {
+            return Response.status(Status.FORBIDDEN)
+                .entity(body.add("message", "The user requested to be logged out is not the user currently logged in"))
+                .build();
         }
+        doLogout();
+
+        if (unauthorize!=null) {
+            // returning 401 UNAUTHORIZED has the nice property that it causes browser (mostly)
+            // to re-prompt for cached credentials to set in the "Authorization: " header to re-login;
+            // TODO however it's not 100%; 
+            // some repeated requests (eg /server/up/extended) in brooklyn webapp seem to keep that header  
+            return Response.status(Status.UNAUTHORIZED)
+                .entity(body.add("message", unauthorize))
+                .build();
+        }
+        
+        return Response.status(Status.OK)
+                .entity(body.add("message", "Logged out user "+currentUser))
+                .build();
     }
 
     private void doLogout() {
+        MultiSessionAttributeAdapter multi = MultiSessionAttributeAdapter.of(req);
+
+        // if we need to intercept session creation then can use this
+        // create TrackingSessionHandler which delegates (no-op if delegate==null esp in setSessionTrackingMode)
+        // and log with stack trace in newHttpSession
+//        HttpServletRequest jreq = req;
+//        if (jreq instanceof ThreadLocalHttpServletRequest) jreq = ((ThreadLocalHttpServletRequest)jreq).get();
+//        if (jreq instanceof ServletRequestWrapper) jreq = (HttpServletRequest) ((ServletRequestWrapper)jreq).getRequest();
+//        if (jreq instanceof Request) {
+//            log.warn("SWAPPING "+MultiSessionAttributeAdapter.info(jreq));
+//            ((Request)jreq).setSessionHandler(new TrackingSessionHandler(((Request)jreq).getSessionHandler()));
+//        } else {
+//            log.warn("UNABLE to swap request"+MultiSessionAttributeAdapter.info(jreq));
+//        }
```

Comment:
Can be removed

Reply:
I'd like to keep for the reasons noted in the comment in this section

```
        // if we need to intercept session creation then can use this
        // create TrackingSessionHandler which delegates (no-op if delegate==null esp in setSessionTrackingMode)
        // and log with stack trace in newHttpSession
```

I know we could restore it from history but unless/until we are really happy with this behaviour I think it is worth making it a little more prominent.

Added

```
        //   can remove this after we've been running happily with new session management model for a while
```